packages/integration-sdk-core/src/data/converters.ts

```diff
@@ -179,11 +179,40 @@ export function getTime(
  * Date(time).getTime()`.
  *
  * @param time a time value
+ * @param precision - the precision of the numeric value being supplied. Used to determine how to convert to ms.
+ *          ms - milliseconds
+ *          sec - seconds
  */
 export function parseTimePropertyValue(
```

Comment:
We don't currently do this, but I think this may be a good place to use overloaded function signatures:
```ts
export function parseTimePropertyValue(time: Date): number | undefined;
export function parseTimePropertyValue(time: number, sourcePrecision: 'ms' | 'sec'): number | undefined
// other overloaded signature here
// source signature
export function parseTimePropertyValue(time: Date | string | number | undefined | null, sourcePrecision?: 'ms' | 'sec'): number | undefined;
```

Note the ability to require that `sourcePrecision` is required when `time` is `number`. This allows prohibit these errors in the future without changing requirements for many already working cases.